sklearn_helpers/search_param.py

```diff
@@ -0,0 +1,119 @@
+
+
+
+
+from sklearn.model_selection import ParameterGrid as grid
+import random
+import numpy as np
+
+
+def fl_search(fun,params:dict,n_iter:int)->dict:
+    """
+    This function is to actively search best parameters for one model, should be better than grid search.
+    parameter space should be discrete and monotonic.
+    :param params: The parameters space to search, with keys as parameters and values are list of candidate values
+    :return: A tuple with first one is the dictionary for the best parameters, and second one is a list as the best score tracking
+    """
+
+
+    keys=list(params.keys())
+
+    num_points={}
+    for key in keys:
+        num_points[key]=len(params[key])
+        params[key].sort()
+
+    lower_point, upper_point=_init_upper_lower_points(keys=keys,num_points=num_points)
+    move_up={}
+    tracking=[]
+
+    i=0
+    while i<n_iter:
```

Comment:
While I realize it's not everyone's cup of tea, this function seems more natural to me as a recursive function where `n_iter` decreases by 1 each time.  Maybe a helper function would be warranted to keep the interface of the function the way it is now.  If we don't want to go recursive, we can still change this from `while i<n_iter` to `for _ in range(n_iter)`.

Reply:
I like the recursive idea, good point!